### PR TITLE
Update S4138 ('prefer-for-of'): Include warning about browser array-like collections

### LINKED ISSUE #3385: Fix FP S4138 ('prefer-for-of'): Check if node does have an iterator to use for-of
The rule S4138 can raise issues when `for...of` is not available for the iterable.

The [typescript-eslint rule implementation](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/src/rules/prefer-for-of.ts) does not check the type, so we could add a extend the [current decorator](https://github.com/SonarSource/SonarJS/blob/master/eslint-bridge/src/linting/eslint/rules/decorators/prefer-for-of-decorator.ts) to filter the flagged nodes for type.

Reported in https://community.sonarsource.com/t/fp-seeking-for-of-where-typescript-prohibits-it/67142

### PR DESCRIPTION
**Pull request by ilia-kebets-sonarsource**
Fixes #3385

### CODE REVIEW

**Comment by ilia-kebets-sonarsource** [comment]
I tried using `!` to affirm that the `rule.meta.messages` props exist, but the QG kicked me down, so I went with optional chaining that fails on coverage.

File: eslint-bridge/src/linting/eslint/rules/decorators/prefer-for-of-decorator.ts [hunk 1]
```diff
@@ -41,6 +42,10 @@ export function decoratePreferForOf(rule: Rule.RuleModule): Rule.RuleModule {
     context.report({
       ...reportDescriptor,
       suggest,
+      messageId: undefined,
+      message:
+        rule.meta?.messages?.preferForOf +
+        ' In a browser environment, `NodeList` and other array-like collections should work by default. If you are using TypeScript and seeing a type error, make sure your configuration is correct.',
```
Comment (yassin-kammoun-sonarsource) [inline eslint-bridge/src/linting/eslint/rules/decorators/prefer-for-of-decorator.ts:48]: I don't think this kind of message should be part of the issue message. It should be moved in the rule description, that is, the RSPEC. Issue messages should be kept as short as possible as well as actionable.

**Review by yassin-kammoun-sonarsource — CHANGES_REQUESTED** [review]
Let's close this PR without merging and open a PR in the RSPEC repository instead to update the rule description.

**Comment by ilia-kebets-sonarsource** [comment]
Closing this as we have modified the message in rspec: https://github.com/SonarSource/rspec/pull/1303

**Pull request closed without merging**